src/main/java/org/jboss/modules/AliasModuleSpec.java

```diff
@@ -22,21 +22,58 @@
 
 package org.jboss.modules;
 
+import java.lang.instrument.ClassFileTransformer;
+import java.security.PermissionCollection;
+import java.util.Collections;
+import java.util.Map;
+
+import org.jboss.modules.filter.PathFilters;
+
 /**
  * A module specification for a module alias.
+ * <p>
+ * Note that because of MODULES-241, alias modules are handled as a regular modules with a single dependency on the alias
+ * target. Use {@link #asConcreteModuleSpec()} to get the regular {@link ConcreteModuleSpec} representing this
+ * {@link AliasModuleSpec}.
  *
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
  */
 public final class AliasModuleSpec extends ModuleSpec {
 
     private final ModuleIdentifier aliasTarget;
+    private final ConcreteModuleSpec concreteModuleSpec;
 
     AliasModuleSpec(final ModuleIdentifier moduleIdentifier, final ModuleIdentifier aliasTarget) {
         super(moduleIdentifier);
         this.aliasTarget = aliasTarget;
+
+        DependencySpec aliasTargetDependency = DependencySpec.createModuleDependencySpec(PathFilters.acceptAll(),
+                PathFilters.acceptAll(), null, aliasTarget, false);
+
+        final String mainClass = null;
+        final AssertionSetting assertionSetting = AssertionSetting.INHERIT;
+        final ResourceLoaderSpec[] resourceLoaders = new ResourceLoaderSpec[0];
```

Comment:
> ideally this would be a constant (would save a couple hundred allocations on server startup).

I'd like to have your eagle eyes!

We actually have such a constant in `ModuleClassLoader` https://github.com/jboss-modules/jboss-modules/blob/c6de7c5483d7331b0a65c226245953aac006db69/src/main/java/org/jboss/modules/ModuleClassLoader.java#L82

From the performance PoV, using `ModuleClassLoader.NO_RESOURCE_LOADERS` here (call it option A) should be equally OK as declaring a new constant in `AliasModuleSpec` (option B) because both compile to `getstatic`.

However making `AliasModuleSpec` dependent on a random class just because it contains a constant we need does not sound right. Moving `NO_RESOURCE_LOADERS` from `ModuleClassLoader` to a more natural place, such as `ResourceLoaderSpec` (option C) would solve it. 

The question is how paranoid do we want to be w.r.t. compatibility with user code? If moving package visible constants around is forbidden, then option B seems to be the best one. Otherwise, C wins.

WDYT?